crates/sui-rpc/src/client/v2/coin_selection.rs

```diff
@@ -25,17 +26,18 @@ impl Client {
     /// Returns an error if there are insufficient funds to meet the required amount or if there is an RPC error
     pub async fn select_coins(
         &self,
-        owner_address: impl Into<String>,
-        coin_type: impl Into<String>,
-        amount: impl Into<u128>,
-        exclude: &[&str],
+        owner_address: impl Into<Address>,
+        coin_type: impl Into<TypeTag>,
+        amount: impl Into<u64>,
```

Comment:
lets just drop the `impl Into<X>` and instead do: `&Address`, `&TypeTag`, `u64`.